### ISSUE TITLE
Feature request: Compression algorithm information

### ISSUE THREAD

**Issue by Mickael-van-der-Beek** [manojkarthick/pqrs#40]
Hello Manoj,

Very useful tool you have built!

One feature I would like to suggest is to display which compression algorithm was used on each column.
Currently, it is possible to see that compression was used based on the size difference of the "total compressed size" and "total uncompressed size" sizes but the actual algorithm used doesn't seem to be displayed.

So the idea would be to show "GZIP", "LZO", "ZSTD", "Brotli", etc. in the schema description table.

**Comment by manojkarthick** [comment]
Thanks for looking into this @SteveLauC - I think it would be best to include the compression algorithm used at a column level in the `pqrs schema --detailed` command's output.

Sample output of `pqrs schema --detailed`:
```
column 0:
--------------------------------------------------------------------------------
column type: INT64
column path: "epochTime"
encodings: PLAIN BIT_PACKED
file path: N/A
file offset: 4
num of values: 1
total compressed size (in bytes): 71
total uncompressed size (in bytes): 69
data page offset: 4
index page offset: N/A
dictionary page offset: N/A
statistics: {min: 1672531499, max: 1672531499, distinct_count: N/A, null_count: 0, min_max_deprecated: false}
```

I think it would be great to add the compression information as another field in this list alongside the "total compressed size".

**Comment by SteveLauC** [comment]
> I think it would be best to include the compression algorithm used at a column level in the `pqrs schema --detailed` command's output.

That would be great

```
Row Group 0:
    xxx: XXX
    xxx: XXX
Row Group 1:
    xxx: XXX
    xxx: XXX
Row Group 2:
    xxx: XXX
    xxx: XXX
...
```
But if there are multiple `row group`s, then the same `column` in different `row group`s can be compressed in different ways, for example:

```
Row Group 0:
     foo: UNCOMPRESSED,
Row Group 0:
    foo: LZO,
```
How can we handle this, only show the algorithms used in the first `row group`?

**Comment by SteveLauC** [comment]
> I think it would be best to include the compression algorithm used at a column level in the `pqrs schema --detailed` command's output.

That would be great, then I will work on it:)

**Comment by SteveLauC** [comment]
Hi @manojkarthick, would you like to get #41 merged first so that we don't need to tackle dependency compatibility problems when implementing the PR for this issue:)

**Comment by SteveLauC** [comment]
I originally thought [parquet#4176](https://github.com/apache/arrow-rs/pull/4176) could be included in release `39.0.0`, but I was wrong, so we have to wait for the release of `40.0.0`:(

**Comment by manojkarthick** [comment]
> I originally thought [parquet#4176](https://github.com/apache/arrow-rs/pull/4176) could be included in release `39.0.0`, but I was wrong, so we have to wait for the release of `40.0.0`:(

@SteveLauC Maybe you want to use the git revision that includes this change in `Cargo.toml` and raise a PR while we wait for this to get merged? It could be updated to `40.0.0` whenever that gets released.
